File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        low=0
        high=0
        total=0
        ans=len(nums)+1
        while high<len(nums):
            total=total+nums[high]
            while total>=target:
                ans=min(ans,high-low+1)
                total=total-nums[low]
                low+=1
            high+=1
        if ans==len(nums)+1:
            return 0
        return ans        

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int low=0,high=0,res=INT_MAX,sum=0;
        for(high=0;high<nums.size();high++)
        {
            sum=sum+nums[high];
            while(sum>=target)
            {
                int len=high-low+1;
                 res=min(res,len);
                sum=sum-nums[low];
                low++;

            }
           

        }
        if(res==INT_MAX)
        return 0;
       return res;
        
    }

};